apps/core/tests/ui/desktop/e2e/account.spec.ts

```diff
@@ -6,3 +6,44 @@ test('Account access is restricted for guest users', async ({ page }) => {
   await expect(page.getByRole('heading', { name: 'My Account' })).toBeVisible({ visible: false });
   await expect(page.getByRole('heading', { name: 'Log in' })).toBeVisible();
 });
+
+test('My Account tabs are displayed and clickable', async ({ page }) => {
+  await page.goto('/login/');
+  await page.getByLabel('Login').click();
+  await page.getByLabel('Email').fill(process.env.TEST_ACCOUNT_EMAIL || '');
+  await page.getByLabel('Password').fill(process.env.TEST_ACCOUNT_PASSWORD || '');
+  await page.getByRole('button', { name: 'Log in' }).click();
+  await page.getByRole('heading', { name: 'My Account' }).waitFor();
```

Comment:
We don't actually need to add `waitFor` because playwright implicilty polls when we do `toBeVisible` unless we want to waitFor something that we think will take extra time than the standard timeout 🤔 

https://playwright.dev/docs/testing-library#replacing-waitfor

Reply:
I would personally favor towards using just `await expect(page.getByRole('heading', { name: 'My Account' })).toBeVisible();` like the rest as it implicitly polls and gives you the bonus assertion you want to have but I wouldn't stop you on that @bc-azvierieva

Reply:
That documentation is for replacing `@testing-library/*` implementation for playwright so not sure that applies here. The `.waitFor` method is basically the same as `await expect(page.getByRole(...)).toBeVisible()`, it's just abstracted to another API. The `.waitFor` defaults to `visible` state: https://playwright.dev/docs/api/class-locator#locator-wait-for-option-state

Reply:
@chanceaclark as for me - there are different use cases for `expect...toBeVisible()` and `waitFor()`
`expect...toBeVisible()` - for asserting something which is the purpose of having test
`waitFor()` - for intermediate elements we navigate through to reach the purpose of test

So I would say,`waitFor()` might be used occasionally rather than instead `expect...toBeVisible()` :)

Reply:
> The `.waitFor` method is basically the sam

Yeah, sorry wrong link but my view on that holds. Unless if we are looking for specific state or waiting for a transition to happen before continuing the test like @bc-azvierieva said, I don't think it's necessary to use. `waitForVisible` or other similar assertions implicitly wait so, I'll make use of them instead. Definitely not opposed to using it where it's valid ☮️ .

Reply:
@bc-azvierieva that's my thought process too. I definitely should've clarified that before, as we arrange a lot of our tests using `expect...toBeVisible()` instead of reserving it for the success criteria of a test.